cam/cam.go

```diff
@@ -164,39 +157,15 @@ func newvideostore(
 	if newConf.Video.Format != "" {
 		format = newConf.Video.Format
 	}
-
-	if newConf.Properties.Width == 0 && newConf.Properties.Height == 0 {
-		vs.logger.Info("received unspecified frame width and height, fetching frame to get dimensions")
-		for range make([]struct{}, numFetchFrameAttempts) {
-			frame, err := camera.DecodeImageFromCamera(ctx, rutils.MimeTypeJPEG, nil, vs.cam)
-			if err != nil {
-				vs.logger.Warn("failed to get and decode frame from camera, retrying. Error: ", err)
-				time.Sleep(retryInterval * time.Second)
-				continue
-			}
-			bounds := frame.Bounds()
-			newConf.Properties.Width = bounds.Dx()
-			newConf.Properties.Height = bounds.Dy()
-			vs.logger.Infof("received frame width and height: %d, %d", newConf.Properties.Width, newConf.Properties.Height)
-			break
-		}
-	}
-	if newConf.Properties.Width == 0 && newConf.Properties.Height == 0 {
-		return nil, fmt.Errorf("failed to get source camera width and height after %d attempts", numFetchFrameAttempts)
-	}
-
-	if newConf.Properties.Framerate == 0 {
-		newConf.Properties.Framerate = defaultFramerate
+	if newConf.Framerate > 0 {
+		vs.framerate = newConf.Framerate
```

Comment:
move the setting of `vs.framerate` above this line - I also think validation should just fail if you set a framerate that's zero or negative - a user has obviously done something wrong for this driver  and we should tell them rather than have it silently build itself.